mqtt/mqtt-broker/tests/common.rs

```diff
@@ -0,0 +1,179 @@
+use std::sync::atomic::{AtomicU32, Ordering};
+
+use futures_util::FutureExt;
+use futures_util::StreamExt;
+use lazy_static::lazy_static;
+use tokio::net::ToSocketAddrs;
+use tokio::sync::{
+    mpsc::{self, Receiver},
+    oneshot::{self, Sender},
+};
+use tokio::task::{JoinError, JoinHandle};
+
+use mqtt3::{
+    proto::{Publication, SubscribeTo},
+    Client, Event, PublishError, PublishHandle, ShutdownError, ShutdownHandle,
+    UpdateSubscriptionError, UpdateSubscriptionHandle,
+};
+use mqtt_broker::{Authenticator, Authorizer, Broker, BrokerState, Error, Server};
+
+/// A wrapper on the mqtt3::Client to help simplify client event loop management.
+#[derive(Debug)]
+pub struct TestClient {
+    publish_handle: PublishHandle,
+    subscription_handle: UpdateSubscriptionHandle,
+    shutdown_handle: ShutdownHandle,
+    pub events_receiver: Receiver<Event>,
```

Comment:
can we make TestClient to implement Stream? It will mean that we can read messages from a client like this:
`client.next().await` and moreover will give us some stream extensions:
`let all_events = client.collect().await` etc...

implementation itself should be pretty straitforward:
```rust
impl Stream for TestClient {
  fn poll_next(&mut self)  -> {
    self.event_receiver.poll_next();
  }
}

Reply:
It is a good idea, but I'd rather explore that in a separate PR for now.